Deploy downcast<> in HTMLCanvasElement

AlexsanderDamaceno opened the pull request:
#### 851fa19d29854477fff502200a1d9099924ce8cf
<pre>
Deploy downcast&lt;&gt; in HTMLCanvasElement
<a href="https://bugs.webkit.org/show_bug.cgi?id=281474">https://bugs.webkit.org/show_bug.cgi?id=281474</a>

Reviewed by Chris Dumez.

 Deploy downcast&lt;&gt; in HTMLCanvasElement

* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::createContext2d):
(WebCore::HTMLCanvasElement::getContext2d):
(WebCore::HTMLCanvasElement::getContextBitmapRenderer):
(WebCore::HTMLCanvasElement::createContextWebGPU):
(WebCore::HTMLCanvasElement::getContextWebGPU):
(WebCore::root):

Canonical link: <a href="https://commits.webkit.org/285198@main">https://commits.webkit.org/285198@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/65f52e89a390af1cd5879f23b09c8019303d58bc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71776 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51189 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24550 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75891 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22981 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/73891 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58990 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22801 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56657 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15155 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74842 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46433 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61829 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37111 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43094 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19296 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21322 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65001 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19659 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77610 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16010 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18843 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/64379 "Found 35 new test failures: fast/editing/document-leak-altered-text-field.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb10_a2-rgba-unsigned_int_2_10_10_10_rev.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb16f-rgb-float.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb16f-rgb-half_float.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb32f-rgb-float.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb565-rgb-unsigned_byte.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb565-rgb-unsigned_short_5_6_5.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb8-rgb-unsigned_byte.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb8ui-rgb_integer-unsigned_byte.html webgl/2.0.y/conformance2/textures/canvas/tex-3d-rgb9_e5-rgb-float.html ... (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16054 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61862 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64391 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15876 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12559 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6199 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46989 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1768 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48060 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49344 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47802 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/af585a6a8caa9ba0826684d38408636620219980. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/35198

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0e602515dd434c411372899e12ff9649f34eb0df. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/35198

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/65f52e89a390af1cd5879f23b09c8019303d58bc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/35198

AlexsanderDamaceno committed:
Deploy downcast<> in HTMLCanvasElement
https://bugs.webkit.org/show_bug.cgi?id=281474

Reviewed by Chris Dumez.

 Deploy downcast<> in HTMLCanvasElement

* Source/WebCore/html/HTMLCanvasElement.cpp:
(WebCore::HTMLCanvasElement::createContext2d):
(WebCore::HTMLCanvasElement::getContext2d):
(WebCore::HTMLCanvasElement::getContextBitmapRenderer):
(WebCore::HTMLCanvasElement::createContextWebGPU):
(WebCore::HTMLCanvasElement::getContextWebGPU):
(WebCore::root):

Canonical link: https://commits.webkit.org/285198@main

webkit-commit-queue commented:
Committed 285198@main (851fa19d2985): <https://commits.webkit.org/285198@main>

Reviewed commits have been landed. Closing PR #35198 and removing active labels.
